server/models/Collective.js

```diff
@@ -466,23 +467,39 @@ function defineModel() {
         },
       },
 
+      /**
+       * @deprecated Keeping this one as a virtual field for easier migration. It should be removed
+       * when we'll move to a dedicated [SocialLinks table](https://github.com/opencollective/opencollective/issues/5097)
+       */
       githubHandle: {
-        type: DataTypes.STRING,
-        set(githubHandle) {
-          if (!githubHandle || githubHandle.length === 0) {
-            this.setDataValue('githubHandle', null);
-            return;
+        type: DataTypes.VIRTUAL,
+        get() {
+          return getGithubHandleFromUrl(this.repositoryUrl);
+        },
+        set(input) {
+          const cleanInput = input?.trim();
+          const githubUrl = getGithubUrlFromHandle(cleanInput);
+          if (githubUrl) {
+            this.setDataValue('repositoryUrl', githubUrl);
+          } else {
+            this.setDataValue('repositoryUrl', null);
           }
+        },
+      },
 
-          // Try to parse github URL, fallback on regular string
-          const githubUrlRegex = /https?:\/\/github\.com\/([^/\s]+)(\/([^/\s]+))?/;
-          const regexResult = githubHandle.match(githubUrlRegex);
-          if (regexResult) {
-            const [, username, , repository] = regexResult;
-            const formattedHandle = repository ? `${username}/${repository}` : username;
-            this.setDataValue('githubHandle', formattedHandle);
+      repositoryUrl: {
+        type: DataTypes.STRING,
+        validate: {
+          notEmpty: true,
+          isUrl: {
+            msg: 'Repository URL must be a valid URL',
+          },
```

Comment:
Don't we have to do any generic length checking on the url here? Just like we do for the githubHandle ? 🤔

Reply:
The repository URL can be a Github URL, a Gitlab URL or anything self-hosted. We can't really enforce strict rules here, apart from the default `2083` length limit from `isUrl`